### DISCUSSION TITLE
Why is UNet2DModel not a SINGLE_FILE_LOADABLE_CLASSES?

### DISCUSSION THREAD

**Discussion by Phlip79** [huggingface/diffusers#8745] [Q&A]
[SINGLE_FILE_LOADABLE_CLASSES](https://github.com/huggingface/diffusers/blob/main/src/diffusers/loaders/single_file_model.py#L48) includes `UNet2DConditionModel` but not `UNet2DModel`. Could someone please explain to me why?

**Comment by sayakpaul** [comment]
Because we don't have any single-file checkpoint for that model class that is sufficiently used enough.

**Comment by Phlip79** [comment]
Can the [implementation functions](https://github.com/huggingface/diffusers/blob/main/src/diffusers/loaders/single_file_model.py#L52-L59) for `UNet2DConditionModel` also be used for `UNet2DModel`?

If so, I believe it would be a rather straightforward change to add support for `UNet2DModel`. I would be happy to make that change.

**Reply by sayakpaul** [reply]
Can you show us a single-file checkpoint for `UNet2DModel` before that?
